sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/LockRenewalOperation.java

```diff
@@ -0,0 +1,212 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.messaging.servicebus;
+
+import com.azure.core.util.CoreUtils;
+import com.azure.core.util.logging.ClientLogger;
+import com.azure.messaging.servicebus.implementation.MessageUtils;
+import com.azure.messaging.servicebus.models.LockRenewalStatus;
+import reactor.core.Disposable;
+import reactor.core.Disposables;
+import reactor.core.publisher.EmitterProcessor;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.FluxSink;
+import reactor.core.publisher.Mono;
+import reactor.core.publisher.MonoProcessor;
+
+import java.time.Duration;
+import java.time.Instant;
+import java.util.Objects;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.function.Function;
+
+/**
+ * Represents a renewal session or message lock renewal operation that.
+ */
+public class LockRenewalOperation implements AutoCloseable {
+    private final ClientLogger logger = new ClientLogger(LockRenewalOperation.class);
+    private final AtomicBoolean isDisposed = new AtomicBoolean();
+    private final AtomicReference<Instant> lockedUntil = new AtomicReference<>();
+    private final AtomicReference<Throwable> throwable = new AtomicReference<>();
+    private final AtomicReference<LockRenewalStatus> status = new AtomicReference<>(LockRenewalStatus.RUNNING);
+    private final MonoProcessor<Void> cancellationProcessor = MonoProcessor.create();
+
+    private final String lockToken;
+    private final boolean isSession;
+    private final Function<String, Mono<Instant>> renewalOperation;
+    private final Disposable subscription;
+
+    /**
+     * Creates a new lock renewal operation. The lock is initially renewed.
+     *
+     * @param lockToken Lock or session id to renew.
+     * @param maxLockRenewalDuration The maximum duration this lock should be renewed.
+     * @param isSession Whether the lock represents a session lock or message lock.
+     * @param renewalOperation The renewal operation to call.
+     */
+    LockRenewalOperation(String lockToken, Duration maxLockRenewalDuration, boolean isSession,
+        Function<String, Mono<Instant>> renewalOperation) {
+        this(lockToken, maxLockRenewalDuration, isSession, renewalOperation, Instant.now());
+    }
```

Comment:
Instant.now() :  This would mean, renew until , current time. is this right ?

Reply:
Yes. This is correct. It'll force a lock renewal.